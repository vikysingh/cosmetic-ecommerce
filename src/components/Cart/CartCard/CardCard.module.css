#CART_CARD {
    width: 70%;
    height: 200px;
    align-items: flex-start; 
    padding-left: 10px;
    margin: 5px 0;
}

#CART_CARD img {
    width: 25%;
    height: auto;
    margin-top: auto;
    margin-bottom: auto;
}

#CART_CARD > div {
    width: 75%;
    padding-top: 40px;
    padding-left: 20px;
}

#CART_CARD h4 {
    font-size: 1rem;
    margin: 0;
    margin-bottom: 10px;
    width: 80%;
}

#CART_CARD a {
    color: #000;
}

#CART_CARD h5 {
    margin-top: 10px;
    font-size: 0.9rem;
}

@media (max-width: 600px) {
    #CART_CARD {
        width: 95%;
        height: 180px;
    }

    #CART_CARD > div {
        width: 70%;
        padding-top: 20px;
        padding-left: 10px;
    }

    #CART_CARD h4 {
        width: 95%;
        margin-bottom: 5px;
    }
}