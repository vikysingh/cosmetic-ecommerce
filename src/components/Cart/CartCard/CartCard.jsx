import React from "react"

import { Typography } from "antd"
import { Link } from "react-router-dom"

import styles from "./CardCard.module.css"
import globalStyles from "../../../styles/Globals.module.css"

const { Title } = Typography

function CartCard({ imgUrl, name, price, id, dispatch }) {

    return <div id={styles.CART_CARD} className={globalStyles.flexRowStartCenter} >
        <img src={imgUrl} alt="" />
        <div>
            <Title level={4}>
                <Link to={`product/${id}`} > {name} </Link>
            </Title> 
            
            <h5> Total: â‚¬{price} </h5>
        </div>
    </div>
}

export default React.memo(CartCard)