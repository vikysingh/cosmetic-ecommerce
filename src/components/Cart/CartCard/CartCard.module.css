.CART_CARD {
    width: 70%;
    height: 200px;
    justify-content: flex-start !important;
    align-items: flex-start !important; 
    padding-left: 10px;
    margin: 5px 0;
}

.CART_CARD img {
    width: 25%;
    height: auto;
    margin-top: auto;
    margin-bottom: auto;
}

.CART_CARD > div {
    width: 75%;
    padding-top: 40px;
    padding-left: 20px;
}

.CART_CARD h5 {
    margin-top: 10px;
    font-size: 0.9rem;
}

.CART_CARD__delete_icon {
    color: red;
    cursor: pointer;
}

@media (max-width: 600px) {
    .CART_CARD {
        width: 95%;
        height: 180px;
    }

    .CART_CARD > div {
        width: 70%;
        padding-top: 20px;
        padding-left: 10px;
    }

    .CART_CARD h4 {
        width: 95%;
        margin-bottom: 5px;
    }
}