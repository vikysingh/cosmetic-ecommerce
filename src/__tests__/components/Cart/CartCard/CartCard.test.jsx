import React from "react"
import { shallow } from "enzyme"

import CartCard from "../../../../components/Cart/CartCard/CartCard"

import { dummyCartData } from "../../../../test_files/dummyTestData"

const { imgUrl, name, price,
    id, dispatch, action, quantity } = dummyCartData

test('Should render CartCard from \nCart', () => {
    const wrapper = shallow(<CartCard imgUrl={imgUrl} name={name}
        price={price} id={id} dispatch={dispatch} action={action} quantity={quantity} />)

    expect(wrapper).toMatchSnapshot()
})