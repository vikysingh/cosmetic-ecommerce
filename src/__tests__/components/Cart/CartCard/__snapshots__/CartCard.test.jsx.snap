// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CartCard from 
Cart 1`] = `
<div
  className="flexRowStartCenter"
  id="CART_CARD"
>
  <img
    alt=""
    src="some url"
  />
  <div>
    <Title
      level={4}
    >
      <[object Object]
        to="/product/49"
      >
         
        Product name
         
      </[object Object]>
    </Title>
    <h5>
       Total: â‚¬
      14.99
       
    </h5>
  </div>
</div>
`;
